### DISCUSSION TITLE
Option to not throw when trying to retrieve an inactive match

### DISCUSSION THREAD

**Discussion by tobyzerner** [TanStack/router#1697] [Ideas]
Currently hooks which accept a `from` option like `useMatch`, `useLoaderData`, `useParams` etc. are automatically invariant and will throw if you try to access a route that is not currently active.

I think it would be useful to be able to disable this behaviour and return `undefined` instead. I know that you can pass `strict: false` to `useParams` to get a union of all possible params, but this does not work for `useMatch` and `useLoaderData` as far as I can tell - these seem to just return the result from the *nearest* matching route instead.

My use-case is that I have a layout route which displays some status based on the params/loader data of a child route. If the child route is not active, that's fine, I can handle that case - there is no need to throw.

In other words it would be nice to be able to do something like this:

```ts
function Layout() {
    const org = useLoaderData({ from: '/_layout/org/$orgId', throw: false });
    //    ^ Organization | undefined
}
```

**Comment by Rick-Phoenix** [comment]
+1 on this. It is hard to use params or let alone data from routes because they throw if there isn't a match even for a millisecond.
And at the same time, these methods cannot be used conditionally because they are hooks.

**Reply by schiller-manuel** [reply]
we have shouldThrow for useMatch. could well be added to the other hooks as well

**Reply by Rick-Phoenix** [reply]
Agreed!

**Reply by schiller-manuel** [reply]
I added `shouldThrow` to `useSearch` and `useParams` in https://github.com/TanStack/router/pull/3368

feel free to replicate this for the remaining hooks

**Reply by Rick-Phoenix** [reply]
Amazing, thank you!
